testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/AudienceTest.java

```diff
@@ -0,0 +1,221 @@
+                .role("account", "view-profile")
+                .role("service-client", "role1")
+                .build();
+        testRealm.getUsers().add(user);
+    }
+
+
+    @Test
+    public void testAudienceProtocolMapperWithClientAudience() throws Exception {
+        // Add audience protocol mapper to the clientScope "audience-scope"
+        ProtocolMapperRepresentation audienceMapper = ProtocolMapperUtil.createAudienceMapper("audience mapper", "service-client",
+                null, true, false);
+        ClientScopeResource clientScope = ApiUtil.findClientScopeByName(testRealm(), "audience-scope");
+        Response resp = clientScope.getProtocolMappers().createMapper(audienceMapper);
+        String mapperId = ApiUtil.getCreatedId(resp);
+        resp.close();
+
+        // Login and check audiences in the token (just accessToken contains it)
+        oauth.scope("openid audience-scope");
+        oauth.doLogin("john", "password");
+        EventRepresentation loginEvent = events.expectLogin()
+                .user(userId)
+                .assertEvent();
+        Tokens tokens = sendTokenRequest(loginEvent, userId,"openid audience-scope", "test-app");
+        // TODO: Frontend client itself should not be in the audiences of access token. Will be fixed in the future
+        assertAudiences(tokens.accessToken, "test-app", "service-client");
+        assertAudiences(tokens.idToken, "test-app");
+
+        // Revert
+        clientScope.getProtocolMappers().delete(mapperId);
+    }
+
+
+    @Test
+    public void testAudienceProtocolMapperWithCustomAudience() throws Exception {
+        // Add audience protocol mapper to the clientScope "audience-scope"
+        ProtocolMapperRepresentation audienceMapper = ProtocolMapperUtil.createAudienceMapper("audience mapper 1", null,
+                "http://host/service/ctx1", true, false);
+        ClientScopeResource clientScope = ApiUtil.findClientScopeByName(testRealm(), "audience-scope");
+        Response resp = clientScope.getProtocolMappers().createMapper(audienceMapper);
+        String mapper1Id = ApiUtil.getCreatedId(resp);
+        resp.close();
+
+        audienceMapper = ProtocolMapperUtil.createAudienceMapper("audience mapper 2", null,
+                "http://host/service/ctx2", true, true);
+        resp = clientScope.getProtocolMappers().createMapper(audienceMapper);
+        String mapper2Id = ApiUtil.getCreatedId(resp);
+        resp.close();
+
+        // Login and check audiences in the token
+        oauth.scope("openid audience-scope");
+        oauth.doLogin("john", "password");
+        EventRepresentation loginEvent = events.expectLogin()
+                .user(userId)
+                .assertEvent();
+        Tokens tokens = sendTokenRequest(loginEvent, userId,"openid audience-scope", "test-app");
+        // TODO: Frontend client itself should not be in the audiences of access token. Will be fixed in the future
```

Comment:
Is it possible to change this behavior somehow and avoid having the frontend in the list of audiences ? Asking because people may want to have in access tokens only the audiences representing the resource servers where token will be used.